### DISCUSSION TITLE
Listview option removes registry key that does not exist

### DISCUSSION THREAD

**Discussion by comdlg32** [valinet/ExplorerPatcher#476] [General]
At https://github.com/valinet/ExplorerPatcher/commit/1c22ebf291eeb77b04f903bbfc59462ce45c53ec it shows that it is removing the registry key `HKEY_CURRENT_USER\Software\Classes\CLSID\{1eeb5b5a-06fb-4732-96b3-975c0194eb39}\InprocServer32`. The issue is that the CLSID is stored in HKEY_CLASSES_ROOT\{1eeb5b5a-06fb-4732-96b3-975c0194eb39}\InProcServer32 and not in HKCU.

**Comment by valinet** [comment]
Issues should be for issues, I said this multiple times. Why not ask first, confirm it's an issue, and then it gets moved here just fine...? Why is following some piece of advice so hard?

The application runs as the logged-in user, unelevated. It does not have access to write to keys in `HKEY_LOCAL_MACHINE`. Thus, any overrides are made using keys located in `HKEY_CURRENT_USER`. As you should probably be aware, usually, something in some path in `HKEY_CURRENT_USER` overrides the same path in `HKEY_LOCAL_MACHINE`. As is the case here, writing in `HKEY_CURRENT_USER\Software\Classes\CLSID\{1eeb5b5a-06fb-4732-96b3-975c0194eb39}\InprocServer32` overrides whatever information is stored in `HKEY_LOCAL_MACHINE\Software\Classes\CLSID\{1eeb5b5a-06fb-4732-96b3-975c0194eb39}\InprocServer32`. By overriding that, I provide a non-working value for `InprocServer32`, and thus the component is not found and `explorer` falls back to using the legacy list view.

Also, specifically for your question, I think: `HKEY_CLASSES_ROOT` is just a "shortcut" to `HKEY_LOCAL_MACHINE\Software\Classes`. They both point to the same data store, which really is `HKEY_LOCAL_MACHINE\Software\Classes`. You can disregard it entirely and refer to it as `HKEY_LOCAL_MACHINE\Software\Classes`, it's the same thing, it's just a convenience shortcut. So, the CLSID is indeed stored in `HKEY_CLASSES_ROOT\CLSID\{1eeb5b5a-06fb-4732-96b3-975c0194eb39}\InProcServer32`, which is the same as `HKEY_LOCAL_MACHINE\Software\Classes\CLSID\{1eeb5b5a-06fb-4732-96b3-975c0194eb39}\InProcServer32` and which I override using entries in `HKEY_CURRENT_USER\Software\Classes\CLSID\{1eeb5b5a-06fb-4732-96b3-975c0194eb39}\InProcServer32`. This is the proper way to do it in my opinion, if one has to do it, not by removing the original information in `HKEY_LOCAL_MACHINE\Software\Classes\CLSID`/`HKEY_CLASSES_ROOT\CLSID` as some guides on matters like this suggest. It is generally good to leave whatever Windows stores in the registry intact, in order to avoid potential issues down the road. Things in `HKEY_CURRENT_USER` you have full control of, plus, if you screw something up and don't remember about it, creating a user account is much easier than reinstalling the entire operating system.

I will close this and move to discussions, as I think it belongs there. Thanks.
